commercial/app/controllers/commercial/SoulmatesController.scala

```diff
@@ -3,37 +3,44 @@ package controllers.commercial
 import model.commercial.soulmates.SoulmatesAgent.{menAgent, newMenAgent, newWomenAgent, womenAgent}
 import model.commercial.soulmates._
 import model.{Cached, NoCache}
+import play.api.libs.json.Json
 import play.api.mvc._
-import play.twirl.api.HtmlFormat
 
 class SoulmatesController extends Controller with implicits.Requests {
 
-  private def result(groupName: String,
-                     view: (Seq[Member], Option[String], Option[String]) => HtmlFormat.Appendable)
-                    (implicit request: Request[AnyContent]): Result = {
+  private def soulmatesSample(groupName: String): Seq[Member] = {
 
-    val sample = {
-      def take3(agent: SoulmatesAgent) = agent.sample().take(3)
-      if (groupName == "mixed") {
+    def take3(agent: SoulmatesAgent) = agent.sample().take(3)
+
+    groupName match {
+      case "mixed" => {
         val members = take3(womenAgent) ++ take3(menAgent)
         Sample.default(members)
-      } else if (groupName == "mixednew") {
+      }
+      case "mixednew" => {
         val members = take3(newWomenAgent) ++ take3(newMenAgent)
         Sample.default(members)
-      } else SoulmatesAgent.sample(groupName)
+      }
+      case _ => SoulmatesAgent.sample(groupName)
     }
+  }
 
-    sample.toList match {
+  def renderSoulmates(groupName: String) = Action { implicit request =>
+    soulmatesSample(groupName).toList match {
       case Nil => NoCache(jsonFormat.nilResult.result)
       case soulmates => Cached(componentMaxAge) {
         val clickMacro = request.getParameter("clickMacro")
         val omnitureId = request.getParameter("omnitureId")
-        jsonFormat.result(view(soulmates, omnitureId, clickMacro))
+        jsonFormat.result(views.html.soulmates.soulmates(soulmates, omnitureId, clickMacro))
       }
     }
   }
 
-  def renderSoulmates(groupName: String): Action[AnyContent] = Action { implicit request =>
-    result(groupName, views.html.soulmates.soulmates(_, _, _))
+  def getSoulmates(groupName: String) = Action { implicit request =>
+
+    val json = Json.toJson(soulmatesSample(groupName))
+    Cached.explicitlyCache(60){
```

Comment:
Okay, in order to cache the result, it needs to be a `CacheableResult`, so would the following work, or is there a better/standard way of otherwise converting a `Result` to a `CacheableResult`?
```
Cached(60.seconds){
      WithoutRevalidationResult(Ok(json))
    }
```
I couldn't find an example where we are already serving cached JSON responses but perhaps my search-fu is just poor this morning. Know of anywhere in the code base we are doing that?